### DISCUSSION TITLE
(amphp dependency) PHP Deprecated:  Optional parameter $exception declared before required parameter $result

### DISCUSSION THREAD

**Discussion by dm-pf** [spatie/laravel-data#624] [Q&A] [answered]
Version 3.11.0 of `spatie/laravel-data` has introduced a new dependency `spatie/php-structure-discoverer` which pulls its own dependency `amphp/amp (v2.6.2)`.

The problem with amp (2.6.2) is that it throws a deprecated warning at PHP >8.0 users:

`PHP Deprecated:  Optional parameter $exception declared before required parameter $result is implicitly treated as a required parameter in /Users/user/myproject/vendor/amphp/amp/lib/functions.php on line 556`

I am aware that this is not a bug IN laravel-data itself, that's why I am not reporting this as a bug. I just hope this message helps other devs as well.

To get rid of this deprecation warning thrown at your users, downgrade laravel-data to:
`"spatie/laravel-data": "3.10.1"`

**Comment by rubenvanassche** [comment, marked as answer]
This should be fixed by now.
